pom.xml

```diff
@@ -732,11 +732,11 @@
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-compiler-plugin</artifactId>
-                    <version>3.5.1</version>
+                    <version>3.6.1</version>
                     <configuration>
                         <encoding>UTF-8</encoding>
-                        <source>1.6</source>
-                        <target>1.6</target>
+                        <source>1.7</source>
+                        <target>1.7</target>
```

Comment:
It is not possible to use source 1.7 and target 1.6. We can use source 1.6 and target 1.6 but then we'd also have to forgo using features that were introduced in 1.7.

Anyway starting with Android 4.4 we can use Java 7 features. Though I'm having a hard time finding a good reference for that. Most of the announcements seem to have rotted away.

https://developer.android.com/guide/platform/j8-jack.html
>Android supports all Java 7 language features and a subset of Java 8 language features that vary by platform version. This page describes the new language features you can use, how to properly configure your project to use them, and any known issues you may encounter.